### PR TITLE
Fix JSON input format.

### PR DESCRIPTION
**Pull request by zheng-da**
*Description of changes:*
A field of a JSON block may contain a list. We need to convert it to numpy array.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/gconstruct/file_io.py [hunk 1]
```diff
@@ -131,9 +141,13 @@ def read_data_json(data_file, data_fields):
         for key in data_fields:
             assert key in record, \
                     f"The data field {key} does not exist in the record {record} of {data_file}."
-            data[key].append(record[key])
+            record1 = np.array(record[key]) if isinstance(record[key], list) else record[key]
+            data[key].append(record1)
     for key in data:
-        data[key] = np.array(data[key])
+        if isinstance(data[key][0], np.ndarray):
```
Comment (classicsong) [inline python/graphstorm/gconstruct/file_io.py:147]: Is it possible that Process 0 has max_len 10 while Process 1 has max_len 12. for example.

File: python/graphstorm/gconstruct/file_io.py [hunk 2]
```diff
@@ -102,6 +102,16 @@ def write_data_csv(data, data_file, delimiter=','):
     data_frame = pd.DataFrame(data)
     data_frame.to_csv(data_file, index=True, sep=delimiter)
 
+def _pad_stack(arrs):
+    max_len = max(len(arr) for arr in arrs)
+    dtype = arrs[0].dtype
+    for i, arr in enumerate(arrs):
```
Comment (classicsong) [inline python/graphstorm/gconstruct/file_io.py:108]: Will this implementation more efficient?
```
new_arrs = np.zeros((len(arr), max_len), dtype=dtype)
for i, arr in enumerate(arrs):
   new_arrs[i][[:len(arr)] = arr
```
So we do not need to allocate many small numpy arrays.

File: python/graphstorm/gconstruct/file_io.py (hunk 1 above)
Comment (zheng-da) [reply python/graphstorm/gconstruct/file_io.py:147]: this is a concern. if this happens, we should return an error.

Comment (zheng-da) [reply python/graphstorm/gconstruct/file_io.py:144]: I'll submit another PR to handle errors in the worker processes.

**Review by classicsong — APPROVED** [review]

**Pull request merged**